src/presenters/admin/indexation-permalink-warning-presenter.php

```diff
@@ -42,22 +61,31 @@ public function present() {
 	}
 
 	/**
-	 * Retrieves the base alert.
+	 * Determines which text to show as reason for the indexation.
 	 *
-	 * @return string The generated alert.
+	 * @param string $reason The saved reason.
+	 *
+	 * @return string The text to show as reason.
 	 */
-	protected function get_base_alert() {
-		$output = '<p>';
-		$output .= \esc_html__( 'Because you changed the category URL setting, some of your SEO data need to be reprocessed.', 'wordpress-seo' );
-		$output .= '</p>';
-		$output .= $this->get_estimate();
-		$output .= \sprintf(
-			'<button type="button" class="button yoast-open-indexation" data-title="<strong>%1$s</strong>" data-settings="yoastIndexationData">%2$s</button>',
-			/* translators: 1: Expands to Yoast. */
-			\sprintf( \esc_html__( '%1$s indexing status', 'wordpress-seo' ), 'Yoast' ),
-			\esc_html__( 'Start processing and speed up your site now', 'wordpress-seo' )
-		);
+	protected function get_text_for_reason( $reason ) {
+		$text = '';
+		switch ( $reason ) {
+			case static::REASON_CATEGORY_BASE_PREFIX :
+				$text = \esc_html__( 'Because of a change in your category URL setting, some of your SEO data need to be reprocessed.', 'wordpress-seo' );
+				break;
 
-		return $output;
+			case static::REASON_PERMALINK_SETTINGS :
```

Comment:
Should this be the default instead? Now there is a possibility the paragraph will be empty if calling this present.

Reply:
I agree. Will add a `default:` clause here.